tests/automated_tests/test_aws_job_execution.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 University of Stuttgart
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""test in-request execution for aws"""
+
+from qunicorn_core.api.api_models.job_dtos import SimpleJobDto, JobRequestDto
+from qunicorn_core.core import job_service
+from qunicorn_core.static.enums.assembler_languages import AssemblerLanguage
+from qunicorn_core.static.enums.job_state import JobState
+from qunicorn_core.static.enums.provider_name import ProviderName
+from tests import test_utils
+from tests.conftest import set_up_env
+
+IS_ASYNCHRONOUS: bool = False
+RESULT_TOLERANCE: int = 100
+
+
+def test_create_and_run_aws_local_simulator():
+    """Tests the create and run job method for synchronous execution of the aws local simulator"""
+    # GIVEN: Database Setup - AWS added as a provider
+    app = set_up_env()
+
+    # WHEN: create_and_run executed
+    with app.app_context():
+        job_request_dto: JobRequestDto = test_utils.get_test_job(ProviderName.AWS)
+        test_utils.save_deployment_and_add_id_to_job(job_request_dto, AssemblerLanguage.QASM3)
+        return_dto: SimpleJobDto = job_service.create_and_run_job(job_request_dto, IS_ASYNCHRONOUS)
+
+        # THEN: Check if the correct job with its result is saved in the db
+        assert return_dto.state == JobState.RUNNING
+
+
+def test_aws_local_simulator_braket_job_results():
+    """creates a new job and a braket deployment from json input (from test resources) and runs it on the aws
+    local_simulator.
+    Then the results in the db are checked."""
+    # GIVEN: Database Setup - AWS added as a provider
+    app = set_up_env()
+    with app.app_context():
+        # WHEN: create_and_run executed in generic_test
```

Comment:
Hätte das Abfragen der Korrektheit glaub nicht ausgelagert sondern hier drin oder in eine eigene Methode gesteckt. Dann bricht hier A das Schema nicht so ganz und B ist auch leichter ersichtlich worauf getestet wird.

Reply:
ja pack einfach das set_up_env auch in die generic_test method. Und dann braucht man in jedem Test nur noch eine Zeile. Kannst dann auch die ganzen extra Kommentare und so entfernen

Reply:
Hab es jetzt mit einer Zeile nur und die Trennung bleibt in der Methode - weil sonst hat man einfach code Duplikate - zumindest wenn man es als 2. Methode mach immer den gleichen Methodenaufruf, aber ja das app = setupenv ist in der generic test mit drin und das with appcontext war eh egal

Reply:
haben jetzt die Aufteilung mit given/when/then dann jetzt mit im generic test